### PR TITLE
Feature/cab 2987

### PR DESCRIPTION
**Pull request by agagne**
The following ticket will add features not included within this work.
https://jira.cac.washington.edu/browse/CAB-2977

### CODE REVIEW

File: src/test/java/edu/uw/edm/contentapi2/service/impl/DocumentFacadeImplTest.java [hunk 1]
```diff
@@ -55,4 +68,57 @@ public void repositoryIsCalledWithCorrectId() throws RepositoryException {
...
+
+
+
+
+    //TODO: add test for searching range
+    //TODO: add test case insensitive
+    //TODO: add test case insensitive with wildcards
+    //TODO: add test with OR conjunction
+    //TODO: add test about other search parameters order, orderBy, pageSize, PageStart
+    //TODO: what about additionalRequestedFields
```
Comment (maximede) [inline src/test/java/edu/uw/edm/contentapi2/service/impl/DocumentFacadeImplTest.java]: ignore. Always return everything

File: src/test/java/edu/uw/edm/contentapi2/service/impl/DocumentFacadeImplTest.java [hunk 2]
```diff
@@ -55,4 +68,57 @@ public void repositoryIsCalledWithCorrectId() throws RepositoryException {
...
+    //TODO: how to handle directDatabaseSearch, should we through an exception or just ignore
```
Comment (maximede) [inline src/test/java/edu/uw/edm/contentapi2/service/impl/DocumentFacadeImplTest.java]: ignore

**Comment by jfshenuw** [comment]
Could some of the functions, such as convertToLucene, be shared between legacy search and regular search?

File: src/main/java/edu/uw/edm/contentapi2/service/util/LegacySearchModelUtils.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+        final String field = searchTerms[0];
+        final String term = convertToLuceneTerm(searchTerms[1], operator);
+        final boolean negate = ((operator == NOT_EQUALS) || (operator == CASE_INSENSITIVE_NOT_EQUALS));
+
+        return new SimpleSearchFilter(field, term, negate);
+    }
+
+    private static String convertToLuceneTerm(String term, String operator) {
+        if (operator == LESS_THAN) {
```
Comment (maximede) [inline src/main/java/edu/uw/edm/contentapi2/service/util/LegacySearchModelUtils.java]: should this be `LESS_THAN.equals(operator)`

**Pull request merged**